### ISSUE TITLE
relayPagination broken if fetching different fields or different counts from the same endpoint

### ISSUE THREAD

**Issue by villesau** [urql-graphql/urql#2797]
### Describe the bug

We made following queries:

```
query MyItemList($first: Int!, $after: String) {
  myItemList(first: $first, after: $after) {
    pageInfo {
      endCursor
      hasNextPage
    }
    edges {
      node {
        ...BasicMyItemInfo
      }
    }
  }
}
```

and:

```
query MyListOther($first: Int!, $after: String) {
  someOtherQuery: myItem(id: "all") {
    ...BasicMyItemInfo
    count
    updatedAt
  }
  myList(first: $first, after: $after) {
    pageInfo {
      endCursor
      startCursor
      hasNextPage
      hasPreviousPage
    }
    edges {
      node {
        ...BasicMyItemInfo
        count
        updatedAt
      }
    }
  }
}

```

When first fetching the list via the first query, everything works, including pagination. When moving to other page in the app which uses the second query, the query returns values correctly in the network tab, but urql fails to deliver the results to the react component. I narrowed it down to the relayPagination: If that was not present, everything worked (except the pagination ofc). We also encountered the issue when using different first -arguments in different queries. This was not enough to fix it, but we also had to fetch the same fields.

You can reproduce the issue by refreshing the "More todos" list. In this case they return different amount of items, but the effect is the same: We get null from relay pagination. I'm not 100% sure if the reproduction is fully correct, but the effect is.

### Reproduction

https://codesandbox.io/s/new-breeze-63inj3?file=/src/components/MoreTodos.js

### Urql version

3.0.3

### Validations

- [X] I can confirm that this is a bug report, and not a feature request, RFC, question, or discussion, for which [GitHub Discussions](https://github.com/FormidableLabs/urql/discussions) should be used
- [X] Read the [docs](https://formidable.com/open-source/urql/docs/).
- [X] Follow our [Code of Conduct](https://github.com/FormidableLabs/urql/blob/main/CODE_OF_CONDUCT.md)

**Comment by kitten** [comment]
I think, this basically comes down to us acknowledging that the helper is simply not meant for production. Basically, as per the comment here: https://github.com/urql-graphql/urql/issues/2916#issuecomment-1377529507

The problem here is that we have to merge data which has different levels of "completeness" and if, once all are merged, the result is not "complete" as per the query, this causes a cache miss. Subsequently, there's no way for Graphcache to know how to resolve this conflict.

I don't think there's a definitive way to resolve this. However, we're currently discussing different ways of how to make these cache misses more obvious, specifically because, right now, mutation results that run into cache misses on a cache read after receiving an API result can experience misses which isn't as obvious as it should be..
In other words, they need to "scream in your face" that something's wrong, ha.

That said, these are the cases where I'd say very specifically: Make Pagination a UI concern, pretty please. It is easier, and it will always yield more predictable results.
Be it, by making some kind of side-effect that stitches lists together, or by using fragmentised components for parts of the list.

Closing as won't fix for now, but open for more discussion here of course :v:
